pkg/apis/apps.kubermatic/v1/types.go

```diff
@@ -37,4 +37,14 @@ const (
 	// ApplicationTypeCNIValue can be used as a value for the ApplicationTypeLabel to indicate that the
 	// application definition / application installation type if CNI (Container Network Interface).
 	ApplicationTypeCNIValue = "cni"
+
+	// ApplicationEnforceAnnotation can be used to mark an ApplicationDefinition as enforced.
+	ApplicationEnforceAnnotation = "apps.kubermatic.k8c.io/enforce"
+
+	// ApplicationDefaultAnnotation can be used to mark an ApplicationDefinition as default.
+	ApplicationDefaultAnnotation = "apps.kubermatic.k8c.io/default"
+
+	// ApplicationTargetDatacentersAnnotation can be used to target known set of datacenters for default/enforced applications
+	// Value is comma separated list of datacenter names.
+	ApplicationTargetDatacentersAnnotation = "apps.kubermatic.k8c.io/target-datacenters"
```

Comment:
Couldn't we extend the ApplicationDefinition CRD accordingly? Why are we relying on annotations for this very central functionality?

Reply:
This has been changed; when I was handed over this ticket a few weeks ago, I was told there had been discussions previously, and we had settled for doing it via labels/annotations. I don't have strong opinions there but I'd prefer using the spec instead of annotations as well. 

I have modified the code to use spec instead, if any of the reviewers is against that I can revert that commit otherwise we just go for it. :)